src/disk.ts

```diff
@@ -0,0 +1,111 @@
+import { platform } from 'os';
+import { ChildProcess, exec } from 'child_process';
+import mkdirp from 'mkdirp';
+
+const HFS_MINIMUM = 1100;
+
+interface DiskHandle {
+  mountPath: string;
+  filePath: string;
+  eject: () => Promise<void>;
+}
+
+function generateDriveName() {
+  return `swae_test_drive_${String(parseInt(String(Math.random() * 1000000)))}`;
+}
+
+function eject(path: string): Promise<void> {
+  const command = platform() === 'darwin' ? `diskutil eject ${path} -force` : `sudo umount ${path}`;
+  const process = exec(command);
```

Comment:
`execa` might be a better choice, as it both provides promise support (and nice helper utils) and cleans-up if the parent crashes first